### PR TITLE
DOCSP-13854 usage examples landing page

### PR DESCRIPTION
**Pull request by kyuan-mongodb**
## Pull Request Info

Added Usage Examples landing page.
This page differs from the the Node and Java driver since we're incorporating environment variables.

### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-13854

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodbcom-staging.corp.mongodb.com/golang/docsworker-xlarge/DOCSP-13854-UsageExamplesLanding/usage-examples/

### Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Does it render on staging correctly?
- [x] Are all the links working?
- [x] Are the staging links in the PR description updated?

### CODE REVIEW

File: source/usage-examples.txt [hunk 1]
```diff
@@ -23,16 +29,10 @@ Overview
...
-- an explanation of the operation in the example showing the
-  purpose and a sample use case for the method
+operations. Each example provides the following:
 
-- an explanation of how to use the operation, including parameters,
-  return values, and common exceptions you might encounter
-
-- a full Go program that you can copy and paste to run the example
-  in your own environment
+- A full Go program that to run in your own environment
```
Comment (melissamahoney-mongodb) [inline source/usage-examples.txt:34]: Change: Typo.
```suggestion
- A full Go program that you can run in your own environment
```

File: source/usage-examples.txt [hunk 2]
```diff
@@ -48,22 +48,49 @@ or you can
...
 :doc:`quick start guide </quick-start>` to learn more about getting
-started with the MongoDB Go Driver. Once you've copied a usage example,
-you'll need to edit the connection string to get the example connected to
-your instance of MongoDB:
+started with the MongoDB Go Driver. 
+
+Once you've copied a usage example, you'll need to edit and place
+:ref:`your connection string <usage-examples-connection-string>` in
+:ref:`your environment variable <usage-examples-env-variable>` to get
+the example connected to your instance of MongoDB.
```
Comment (melissamahoney-mongodb) [inline source/usage-examples.txt:56]: Change: 
- Use the simplest tense. [style guide ref](https://docs.mongodb.com/meta/style-guide/writing/use-effective-verbs/#use-the-simplest-tense)
- I think its understood that they will have copied and pasted (especially given the previous paragraph) and we can simplify this down.

```suggestion
To connect the example to your instance of MongoDB, you must edit
:ref:`your environment variable <usage-examples-env-variable>` to include 
:ref:`your connection string <usage-examples-connection-string>`. 
```

File: source/usage-examples.txt [hunk 3]
```diff
@@ -48,22 +48,49 @@ or you can
 Once you have imported the dataset, you can copy and paste a usage
```
Comment (melissamahoney-mongodb) [inline source/usage-examples.txt:48]: Change: Simplest tense.
```suggestion
Once you import the dataset, you can copy and paste a usage
```

File: source/usage-examples.txt [hunk 4]
```diff
@@ -48,22 +48,49 @@ or you can
...
+Environment Variable
+~~~~~~~~~~~~~~~~~~~~
+
+You can use `GoDotEnv <https://github.com/joho/godotenv>`__ to set up
+your environment variable. 
 
-   // Replace the following with your MongoDB deployment's connection string.
-   uri := "mongodb+srv://<user>:<password>@<cluster-url>?retryWrites=true&w=majority";
+When you get to the "Usage" section, add the following application
+configuration in your ``.env`` file at the root of your project:
```
Comment (melissamahoney-mongodb) [inline source/usage-examples.txt:71]: Change: At first I didn't know which Usage section you were talking about. I think I would include this as additional context instead.

```suggestion
Add the following application configuration in your ``.env`` 
file at the root of your project. For more information, see the
`GoDotEnv documentation <https://github.com/joho/godotenv#usage>`__.
```

File: source/usage-examples.txt [hunk 5]
```diff
@@ -48,22 +48,49 @@ or you can
...
+When you get to the "Usage" section, add the following application
+configuration in your ``.env`` file at the root of your project: 
+
+.. code-block::
+
+   MONGODB_URI=mongodb+srv://<user>:<password>@<cluster-url>?retryWrites=true&w=majority
+
+.. note::
+
+   Replace the following with your MongoDB deployment's connection string.
```
Comment (melissamahoney-mongodb) [inline source/usage-examples.txt:79]: Question: There is nothing following... was this copypasta?

File: source/usage-examples.txt [hunk 6]
```diff
@@ -48,22 +48,49 @@ or you can
...
-``uri`` variable in the usage example. If your instance uses
+:manual:`connection string </reference/connection-string/>` to set the
+``MONGODB_URI`` environment variable in the usage example. If your instance uses
 :manual:`SCRAM authentication </core/security-scram/>`, you can replace
 ``<user>`` with your username, ``<password>`` with your password, and
 ``<cluster-url>`` with the IP address or URL of your instance.
 
 For more information about connecting to your MongoDB instance, see the
-Connection Guide.
+<TODO: Connection Guide>.
```
Comment (melissamahoney-mongodb) [inline source/usage-examples.txt:95]: Question: Just making sure you knew this TODO was here? Don't want it to go out in prod :)

**Review by melissamahoney-mongodb — CHANGES_REQUESTED** [review]
A few changes, mostly for style. :)

File: source/usage-examples.txt (hunk 6 above)
Comment (kyuan-mongodb) [reply source/usage-examples.txt:95]: Yes, there's quite a few throughout at the moment -- we'll be doing a pass to update them before putting the docs in prod :)

File: source/usage-examples.txt (hunk 5 above)
Comment (kyuan-mongodb) [reply source/usage-examples.txt:79]: 😅 I rearranged it and forgot to update it to say "preceding"

File: source/usage-examples.txt [hunk 7]
```diff
@@ -23,47 +29,67 @@ Overview
...
+root of your project. For more information, see the
+`GoDotEnv documentation <https://github.com/joho/godotenv#usage>`__.
+
+.. code-block::
+
+   MONGODB_URI=mongodb+srv://<user>:<password>@<cluster-url>?retryWrites=true&w=majority
+
+.. note::
+
+   Replace the preceding with your MongoDB deployment's connection string.
```
Comment (melissamahoney-mongodb) [inline source/usage-examples.txt:78]: Nit: The preceding what? Maybe "the preceding example connection string"?

**Review by melissamahoney-mongodb — APPROVED** [review]
One nit then LGTM :)

**Pull request merged**